### PR TITLE
Add constructor for FormatOptions that coalesces nothing

### PR DESCRIPTION
**Pull request by non-Jedi**
Any field of FormatOptions with nothing will be set to default value.

This is the part of fixing https://github.com/julia-vscode/LanguageServer.jl/issues/624

### CODE REVIEW

**Comment by non-Jedi** [comment]
Double-checked this while doing the lint one, and currently `lineends` is unspecified by the vscode config so is set to `false` (only one not set to true by default). Should that one be disabled for best default UX?

**Comment by non-Jedi** [comment]
Broken tests appear unrelated?

**Comment by ZacLN** [comment]
Yes I think so

**Comment by davidanthoff** [comment]
@non-Jedi could you merge `master` in? I think I just fixed the tests there.

File: src/DocumentFormat.jl [hunk 1]
```diff
@@ -18,7 +20,11 @@ mutable struct FormatOptions
     lineends::Bool
     kw::Bool
 end
-FormatOptions() = FormatOptions(4, true, true, true, true, true, true, true, true, true, true)
+FormatOptions() = FormatOptions(default_options...)
+
+# Default to 4-space indentation with all options on
```
Comment (non-Jedi) [inline src/DocumentFormat.jl:25]: Crap. I forgot to delete this clearly inaccurate comment. :/

Comment (davidanthoff) [reply src/DocumentFormat.jl:25]: No  worries, follow up PR is fine :) We merge fast here ;)

**Pull request merged**